packages/lib/src/core/Errors/utils.ts

```diff
@@ -45,3 +48,137 @@ export const addErrorTranslationsToObject = (originalObj, i18n) => {
+                } else {
+                    errorMsg = i18n.get(errObj.errorMessage) + SR_INDICATOR_PREFIX; // is ValidationRuleResult || GenericError || an as yet incorrectly formed error
+                }
+            }
+
+            let errorMessage = errorMsg;
+            /**
+             * For some fields we might need to append the field type to the start of the error message (varies on a component by component basis)
+             * - necessary for a11y, when we know the translated error msg doesn't contain a reference to the field it refers to
+             * TODO - in the future this should be something we can get rid of once we align all our error texts and translations
+             */
+            if (fieldTypeMappingFn) {
+                const fieldType: string = fieldTypeMappingFn(key, i18n, countrySpecificLabels); // Get translation for field type
+                if (fieldType) errorMessage = `${fieldType}: ${errorMsg}`;
+            }
+
+            acc.push({ field: key, errorMessage, errorCode });
+
+            if (layout) acc.sort((a, b) => layout.indexOf(a.field) - layout.indexOf(b.field));
+        }
+        return acc;
+    }, []);
+
+    return sortedErrors;
+};
+
+/**
+ * Implemented as a partial, with an object containing the first 6 arguments; then the final argument, errors, is passed to the partial
+ *
+ * NOTE: using this generic error setting fny is only suitable when errors for the SRPanel are *only* generated by showValidation().
+ * When errors are also generated onBlur, as the user leaves the input, the SR message generation becomes more complex - see CardInput as an example
+ */
+export const setSRMessagesFromErrors = ({ i18n, fieldTypeMappingFn, isValidating, SRPanelRef }, errors, layout, countrySpecificLabels) => {
+    const currentErrorsSortedByLayout = sortErrorsByLayout({
+        errors,
+        i18n,
+        fieldTypeMappingFn,
+        countrySpecificLabels,
+        layout
+    });
+
+    const doLog = false;
+
+    if (doLog) console.log('### setSRMessagesFromErrors::currentErrorsSortedByLayout:: ', currentErrorsSortedByLayout);
+
+    if (currentErrorsSortedByLayout.length) {
+        /** If validating i.e. "on submit" type event - then display all errors in the SR panel */
+        if (isValidating.current) {
+            const errorMsgArr: string[] = currentErrorsSortedByLayout.map(errObj => errObj.errorMessage);
+            if (doLog) console.log('### setSRMessagesFromErrors:: #1 multiple errors:: (validating) errorMsgArr=', errorMsgArr);
+            SRPanelRef.setMessages(errorMsgArr);
+
+            // Remove 'showValidation' mode - allowing time for collation of all the fields in error whilst it is 'showValidation' mode (some errors come in a second render pass)
+            setTimeout(() => {
```

Comment:
I think we should look into stopping using Refs for state management. We will run into more and more of these issues.
@ribeiroguilherme what do you think?

Reply:
How is this state management? `SRPanelRef` is an object on which we expose the public method needed to pass the messages array

Reply:
Not `SRPanelRef` but `isValidating`

Reply:
ok. That didn't make it into the code snippet for your comment...
But the problem is surely not with the `isValidating` flag so much as it's a shame we need to set a timeout whilst the render passes, that generate all the errors, settle.
I would also prefer it if there was a better mechanism.

Reply:
Oh, yeah right I "selected" the `setTimeout` line. 
The objective would be to `useState`.
This could be something like:
```
[1. Submit setValidating(true)] -> [2. useEffect(()=>,[isValidate])] -> [3. setSRMessage() => .... setValidating(false)]
1. We set the state on submit
2. The setSRMessage is triggered on by listening on the isValidate
3. setSRMessage sets the isValidate state again
```

Reply:
Food for thought: There is a new way of handling state management in the React ecosystem, they call it [signals](https://preactjs.com/guide/v10/signals/). 

I haven't looked into it myself yet, but I heard it is promising. Perhaps we could take a look into this another time and see if it can help with the problem of using `useRef` to store values.

Reply:
Yes, lets. 
I've tried a useState/useEffect based solution to avoid the setTimeout, but when and where we should set the validating state and how we should react to it changing is complicated, so for now I'm sticking with the original useRef flag